README.md

```diff
@@ -1,15 +1,63 @@
-# TypicalBot
+<p align="center">
+    <h1 align="center">TypicalBot</h1>
+</p>
+<p align="center">
+    <a target="_blank" href="https://discord.gg/typicalbot"><img src="https://discordapp.com/api/guilds/163038706117115906/embed.png?style=shield" alt="Discord"></a>
+    <a target="_blank" href="https://stackshare.io/typicalbot-llc/typicalbot"><img src="https://img.shields.io/badge/tech-stack-0690fa.svg?style=flat" alt="StackShare"></a>
+</p>
 
-This version of TypicalBot is still under development. We will update the readme
-with more information in the future. For those that are wondering how to run the
-bot to contribute, you will need to open two terminals. The first one to run the
-gateway and the second one to run the rest proxy. You may also need more
-terminals if you are interacting with other parts of the bot process (such as
-the database).
+## About TypicalBot
 
-**Run the gateway** `deno run -A src/gateway/mod.ts`
+> **Note:** This repository contains the code to the new version of TypicalBot
+> that is under development. If you are looking for the current version of
+> TypicalBot, visit the main
+> [repository](https://github.com/typicalbot/typicalbot).
 
-**Run the rest proxy** `deno run --unstable -A src/rest/mod.ts`
+A description about TypicalBot
+
+## Table of Contents
+
+- [Ecosystem](#ecosystem)
+- [Getting Started](#getting-started)
+  - [Prerequisites](#prerequisites)
+  - [Installation](#installation)
+- [Security Vulnerabilities](#security-vulnerabilities)
+- [License](#license)
+
+## Ecosystem
+
+| Project                                                                        | Language          | Description                                   |
+| ------------------------------------------------------------------------------ | ----------------- | --------------------------------------------- |
+| [rest proxy](https://github.com/typicalbot/typicalbot-deno/tree/main/src/rest) | TypeScript (Deno) | REST proxy to handle all Discord events.      |
+| [gateway](https://github.com/typicalbot/typicalbot-deno/tree/main/src/gateway) | TypeScript (Deno) | Gateway connection to Discord.                |
+| [internal-api](https://github.com/typicalbot/typicalbot-internal-api)          | Rust              | REST api to handle all calls to the database. |
+
+## Getting Started
+
+This section provides a quick-start guide
+
+### Prerequisites
+
+- [Deno](https://deno.land/): Deno 11.1 or newer is required.
+- [Rust](https://www.rust-lang.org/): Rust 1.54 or newer is required
+- [MongoDB](https://www.mongodb.com/): MongoDB 4.x or newer is required
+
+### Installation
+
+1. Ensure all prerequisites are installed.
+2. Clone the repository, ie.
+   `git clone https://github.com/typicalbot/typicalbot-deno.git`
+3. Cache the dependencies, ie. `deno cache deps.ts`
+4. Create an `.env` file using the example file, fill out the missing variables.
+5. Run `deno run --unstable -A src/rest/mod.ts` to start the REST proxy.
```

Comment:
`denox` can also be installed and then the REST proxy can be started using `denox run start:rest`.